WooCommerce/Classes/POS/Models/CartItem+Order.swift

```diff
@@ -0,0 +1,41 @@
+import struct Yosemite.Order
+import struct Yosemite.OrderItem
+import protocol Yosemite.POSItem
+
+extension CartItem {
+    /// Checks if the order has the same items as the cart does.
+    /// Used in POS to check if we need to resync the Order with cart items
+    /// - Returns: Whether the order has the same items as the cart does
+    static func areOrderAndCartDifferent(order: Order?, cartItems: [CartItem]) -> Bool {
+        guard let order else {
+            return cartItems.isNotEmpty
+        }
+        // first we get list of all products 1 by 1
+        var singleQuantityOrderItems: [OrderItem] = order.items.flatMap {
```

Comment:
nitpick;

Theoretically, we could compare dictionaries for efficiency to avoid sorting but likely not worth it if we want to expand the solution since the key would need to be unclear.

```
        let orderItemsCount = order.items.reduce(into: [:]) { counts, item in
            counts[item.productID, default: 0] += item.quantity.intValue
        }
        let cartItemsCount = cartItems.reduce(into: [:]) { counts, item in
            counts[item.item.productID, default: 0] += item.quantity
        }

        return orderItemsCount != cartItemsCount
  ```